cli/tests/test_rebase_command.rs

```diff
@@ -267,37 +267,41 @@ fn test_rebase_branch_with_merge() {
 
     create_commit(&test_env, &repo_path, "a", &[]);
     create_commit(&test_env, &repo_path, "b", &["a"]);
-    create_commit(&test_env, &repo_path, "c", &[]);
-    create_commit(&test_env, &repo_path, "d", &["c"]);
-    create_commit(&test_env, &repo_path, "e", &["a", "d"]);
+    create_commit(&test_env, &repo_path, "c", &["a"]);
+    create_commit(&test_env, &repo_path, "d", &[]);
+    create_commit(&test_env, &repo_path, "e", &["d"]);
+    create_commit(&test_env, &repo_path, "f", &["c", "e"]);
     // Test the setup
     insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
-    @    e
+    @    f
     ├─╮
+    │ ◉  e
     │ ◉  d
-    │ ◉  c
+    ◉ │  c
     │ │ ◉  b
     ├───╯
     ◉ │  a
     ├─╯
     ◉
     "###);
 
-    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["rebase", "-b", "d", "-d", "b"]);
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["rebase", "-b", "e", "-d", "b"]);
     insta::assert_snapshot!(stdout, @"");
     insta::assert_snapshot!(stderr, @r###"
     Rebased 3 commits
-    Working copy now at: znkkpsqq 5f8a3db2 e | e
-    Parent commit      : rlvkpnrz 2443ea76 a | a
-    Parent commit      : vruxwmqv 1677f795 d | d
+    Working copy now at: kmkuslsw 29e82c87 f | f
+    Parent commit      : royxmykx c0cb3a0b c | c
+    Parent commit      : znkkpsqq 6b5a695a e | e
     Added 1 files, modified 0 files, removed 0 files
     "###);
+    // TODO: Commit c should also have been rebased
```

Comment:
I'm not against the new behavior, but I don't think `c` belongs to the "--branch" of `e` (which I think is `::e`.)

Reply:
You're saying we should rename it to `-t/--tree` or something? (`--reachable` would make a lot of sense, but `-r` is of course a bad short name)

Reply:
Yes, but I don't have a better name starting with "b". Both `--branch`/`--base` suggest that the argument is the head, I think.

Reply:
And `--base` sounds a lot more like what `--source` does anyway IMO :)

Reply:
Btw, already before this PR, the "branch" definition isn't Git-like. Let's say you have this history:
```
C
|
| B
|/
A
```
If you do `jj rebase -b B -d X`, we're also going to rebase `C`.

Reply:
I think that's because jj by default auto-rebases descendants. `C` is rebased as a result of rebased `A`. If `C` had other ancestors, they wouldn't be rebased onto `X`.

> And `--base` sounds a lot more like what `--source` does anyway IMO :)

True. I read `--base` as `--base-of` just because Mercurial behaves that way.

Reply:
> I think that's because jj by default auto-rebases descendants. `C` is rebased as a result of rebased `A`. If `C` had other ancestors, they wouldn't be rebased onto `X`.

Mercurial does the same, though (I just verified).